### PR TITLE
[Distributed] Improve error messages for protocol conformances

### PR DESCRIPTION
**Pull request by ktoso**
Include the 'distributed method' wording when we report conformance errors about distributed methods.

### CODE REVIEW

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -4363,7 +4363,7 @@ NOTE(note_add_distributed_to_decl,none,
      "add 'distributed' to %0 to make this %1 witness the protocol requirement",
      (DeclName, DescriptiveDeclKind))
 NOTE(note_distributed_requirement_defined_here,none,
-     "distributed function requirement %0 declared here",
+     "distributed instance method requirement %0 declared here",
```
Comment (ktoso) [inline include/swift/AST/DiagnosticsSema.def:4366]: consistently fixed "distributed function" to "distributed instance method" that's more correct right?

File: include/swift/AST/DiagnosticsSema.def [hunk 2]
```diff
@@ -4502,13 +4502,13 @@ NOTE(note_distributed_actor_isolated_method,none,
...
-      "only 'distributed' functions can be called on a potentially remote distributed actor",
+      "only 'distributed' instance methods can be called on a potentially remote distributed actor",
       ())
 ERROR(distributed_actor_func_param_not_codable,none,
-      "distributed function parameter '%0' of type %1 does not conform to 'Codable'",
+      "distributed instance method parameter '%0' of type %1 does not conform to 'Codable'",
       (StringRef, Type))
 ERROR(distributed_actor_func_result_not_codable,none,
-      "distributed function result type %0 does not conform to 'Codable'",
+      "distributed instance method result type %0 does not conform to 'Codable'",
```
Comment (ktoso) [inline include/swift/AST/DiagnosticsSema.def:4511]: this sounds right now

File: include/swift/AST/DiagnosticsSema.def [hunk 3]
```diff
@@ -4519,6 +4519,9 @@ ERROR(nonisolated_distributed_actor_storage,none,
 ERROR(distributed_actor_func_nonisolated, none,
       "function %0 cannot be both 'nonisolated' and 'distributed'",
       (DeclName))
+ERROR(distributed_actor_func_private, none,
+      "%0 %1 cannot be 'private'",
+      (DescriptiveDeclKind, DeclName))
```
Comment (ktoso) [inline include/swift/AST/DiagnosticsSema.def:4524]: we now reject `distributed private func` as it would not be invocable.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test windows Platform

**Review by xedin — APPROVED** [review]
I'd suggest to replace all of the hard-coded `distributed instance method` in .def file with `DescriptiveDeclKind` argument, that way in the future you'd just need to change the next in one place.

**Comment by ktoso** [comment]
Yeah you're right, I'll keep cleaning this up -- will do more of that in today's followup PRs I think right away.

**Pull request merged**